shuup/campaigns/models/basket_conditions.py

```diff
@@ -209,3 +209,28 @@ def values(self):
     @values.setter
     def values(self, values):
         self.contacts = values
+
+
+class CategoryProductsBasketCondition(BasketCondition):
+    identifier = "basket_category_condition"
+    name = _("Category products in basket")
+
+    operator = EnumIntegerField(
+        ComparisonOperator, default=ComparisonOperator.GTE, verbose_name=_("operator"))
+    quantity = models.PositiveIntegerField(default=1, verbose_name=_("quantity"))
+    category = models.ForeignKey(Category, verbose_name=_("category"), blank=True)
+
+    def matches(self, basket, lines):
+        category_product_ids = self.category.shop_products.values_list("product_id", flat=True)
```

Comment:
number of products in a category will almost always be way larger than the number of items in a basket. IMO, maybe create a list of product ids in the basket, then filter and count the products in category based on that. Just a thought...